WordPress/openverse
# Improve image grid

## Fixes



Fixes #547 by @fcoveram
Fixes #2768 by @zackkrida 
Fixes #2712 by @dhruvkb 
Fixes #563 by @obulat 

## Description


This PR improves the image grid on the image search page, making more images fit in a single row.

Previously, the grid was using a ["padding hack"](https://css-tricks.com/aspect-ratio-boxes/). This PR updates the implementation to use the now widely supported [`aspect-ratio`](https://web.dev/aspect-ratio/).

Updating the image layout from using absolute positioning to CSS grid allowed to fix the padding of the license element on vertical images, as well as to put the license elements below the image on `xs` mobile screens.

The tests were updated to add another set of breakpoints to use: `xs` (instead of `sm`) for mobile and `xl` for desktop. I think that all of the `mobileAndDesktop` should move to this set of tests in the future because `xs` is more representative of the mobile layout than `sm`.

## Screenshots

### `xl`
<img width="1233" alt="Screenshot 2023-09-05 at 6 36 35 AM" src="https://github.com/WordPress/openverse/assets/15233243/9c43be5a-e1d0-448a-8658-c0704cbde560">

### `lg`

<img width="967" alt="Screenshot 2023-09-05 at 6 38 31 AM" src="https://github.com/WordPress/openverse/assets/15233243/58934279-0962-4173-aa39-2c214b134ce7">

### `md`

<img width="747" alt="Screenshot 2023-09-05 at 6 39 10 AM" src="https://github.com/WordPress/openverse/assets/15233243/456ca196-e32e-4a42-88d8-1e9bef2367db">

### `sm`

<img width="621" alt="Screenshot 2023-09-05 at 6 39 56 AM" src="https://github.com/WordPress/openverse/assets/15233243/9a9b4036-486c-4ac5-ae79-23a4b8a7c559">

### `xs`
<img width="356" alt="Screenshot 2023-09-05 at 6 40 54 AM" src="https://github.com/WordPress/openverse/assets/15233243/3e79bf5d-19c0-42e6-a810-05489ebac682">


## Testing Instructions

Run `just install`, `just frontend/run dev:only`.
Search for "cat" images.
Notice that the image grid rows contain more images now. For `xl` screen (1280px), there should be 5-6 images in a row, compared to 4-5 currently.
The smallest screen should show the license icons below the image, even without hovering on the image.
Find the vertical photo of the cat in #2768 and see that the license icons are displayed with correct paddings within the container.

<img width="187" alt="Screenshot 2023-09-05 at 7 13 20 AM" src="https://github.com/WordPress/openverse/assets/15233243/8bc0cf10-4963-45fd-883b-3c2856271a2f">

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
